sympy/functions/elementary/tests/test_complexes.py

```diff
@@ -335,6 +335,10 @@ def test_sign():
     d = p - q
     assert sign(d).func is sign or sign(d) == 0
 
+    # test leading term
+    assert sign(x).as_leading_term(x) == S.One
+    assert sign(-x).as_leading_term(x) == -S.One
```

Comment:
Should test for cases like sign(I*x) or even `sign(x).as....(x, cdir = I)`

Reply:
@arnabnandikgp could you address this. I would like more cases 

sign(sin(x)), sign(x + 2), sign(x + I), sign(I*x) whatever you consider necessary here !